readme.md

# APPMONITOR :: Client Checks #


# Description #

This repository with client checks extends the appmonitor repository. It contains checks for default php web aplications. This prevents to write the same checks for another instance of the same product again.

The client checks exist for

* Concrete 5 (Cms)
* Ilias (LMS)
* Limesurvey (Online survey)
* Matomo (Analytics)


# Requirements #

You need to install the Appmonitor first.



The application monitor is an additional monitoring to the classic monitoring of a servers and its services. It makes checks from the point of view of the application. With its credentials and permissions.
https://github.com/iml-it/appmonitor

The client and server directory is needed for a monitoring server.

The client directory is needed on your application server.


# Installation #

## Install Appmonitor client ##

Install the appmonitor client on your web application.
https://github.com/iml-it/appmonitor

In the appmonitor client folder I expect the following structure:


```
|   (other checks)
|   general_include.php
|
\---classes
        appmonitor-checks.class.php
        appmonitor-client.class.php
```

## Add a custom check ##

Each check consists of 2 files:

* a script *check-[product].php* that contains the logic for checks checks
* a config-sample *check-[product].settings.sample.php* file.

What you need to do is:

- Download Archive of https://github.com/iml-it/appmonitor-clients
- Copy the needed files for your web app into the appmonitor client directory.
- Create a copy of the sample file and set your webroot
- Test json output
- Add https://exaple.com/appmonitor/client/check-[product].php in you appmonitor webgui


## Example: for the CMS Concrete5 ##


Copy check-concrete5* to have these files in the client directory:


```
|   check-concrete5.php
|   check-concrete5.settings.sample.php
|   general_include.php
|   readme.md
|
\---classes
        appmonitor-checks.class.php
        appmonitor-client.class.php
```

Copy *check-concrete5.settings.sample.php* to *check-concrete5.settings.php* (without ".sample").

Edit *check-concrete5.settings.php* and change application root if it is not the webroot.

Open https://exaple.com/appmonitor/client/check-concrete5.php in your webbrowser.
If this OK got to your appmonitor server gui - tab "setup" and and add the same url for a new client check.

history.txt

HISTORY

v0.05  2018-08-27
  - add limesurvey client (requires appmonitor client version 0.52)

v0.04  2018-07-26
  - add ssl check in all client checks if https is used in the request

v0.03  2018-07-16
  - add port in mysql check of ILIAS (after appmonitor client update to 0.42)

v0.02  2018-07-16
  - add ILIAS check

v0.01  2018-07-16
  - first commit with 2 checks from Axel (https://www.axel-hahn.de/)


client/check-ilias.php
<?php
/* ______________________________________________________________________
 * 
 * A P P M O N I T O R  ::  CLIENT - CHECK
 * ______________________________________________________________________
 * 
 * Check for a ILIAS instance.
 * LMS https://www.ilias.de
 * 
 * It checks 
 * - config file must be readably
 * - default client config must be writable
 * - datatadir must be writable
 * - datatadir/[Default-client] must be writable
 * - mysql connect of default client
 * 
 * @author: Axel Hahn
 * ----------------------------------------------------------------------
 * 2018-07-16  v0.02
 * 2018-07-17  v0.03  add port in mysql check
 */

// ----------------------------------------------------------------------
// CONFIG
// ----------------------------------------------------------------------
require_once('classes/appmonitor-client.class.php');
$oMonitor = new appmonitor();
$oMonitor->setWebsite('ILIAS Instance');

@include 'general_include.php';
require_once 'check-ilias.settings.php';

// ----------------------------------------------------------------------
// Read ILIAS specific config items
// ----------------------------------------------------------------------


$sConfigfile = $sApproot . '/ilias.ini.php';
if (!file_exists($sConfigfile)) {
    header('HTTP/1.0 503 Service Unavailable');
    die('ERROR: Config file was not found. Set a correct $sApproot pointing to Ilias install dir.');
}

$aConfig = parse_ini_file($sConfigfile, true);

$sClientConfigFile=(
	isset($aConfig['server']['absolute_path'])
	&& isset($aConfig['clients']['path']) 
	&& isset($aConfig['clients']['default']) 
	&& isset($aConfig['clients']['inifile'])
	) ? $aConfig['server']['absolute_path'].'/'.$aConfig['clients']['path'].'/'.$aConfig['clients']['default'].'/'.$aConfig['clients']['inifile']
	: false
	;

if(!$sClientConfigFile || !file_exists($sClientConfigFile)){
    header('HTTP/1.0 503 Service Unavailable');
    die('ERROR: Cannot parse client config from ['.basename($sConfigfile).'].');	
}

$aClientConfig = parse_ini_file($sClientConfigFile, true);

$aDb=array(
  'server'   => $aClientConfig['db']['host'],
  'username' => $aClientConfig['db']['user'],
  'password' => $aClientConfig['db']['pass'],
  'database' => $aClientConfig['db']['name'],
  'port'     => $aClientConfig['db']['port'],
); 

// ----------------------------------------------------------------------
// checks
// ----------------------------------------------------------------------

$oMonitor->addCheck(
    array(
        "name" => "check config file",
        "description" => "The config file must be readable",
        "check" => array(
            "function" => "File",
            "params" => array(
                "filename" => $sConfigfile,
                "file" => true,
                "readable" => true,
            ),
        ),
    )
);
$oMonitor->addCheck(
    array(
        "name" => "check config file",
        "description" => "The config file must be writable",
        "check" => array(
            "function" => "File",
            "params" => array(
                "filename" => $sClientConfigFile,
                "file" => true,
                "writable" => true,
            ),
        ),
    )
);
$oMonitor->addCheck(
    array(
        "name" => "check dir clients - datadir",
        "description" => "The datadir must be writable",
        "check" => array(
            "function" => "File",
            "params" => array(
                "filename" => $aConfig['clients']['datadir'],
                "dir" => true,
                "writable" => true,
            ),
        ),
    )
);
$oMonitor->addCheck(
    array(
        "name" => "check dir of default client",
        "description" => "The datadir must be writable",
        "check" => array(
            "function" => "File",
            "params" => array(
                "filename" => $aConfig['clients']['datadir'].'/'.$aConfig['clients']['default'],
                "dir" => true,
                "writable" => true,
            ),
        ),
    )
);

$oMonitor->addCheck(
    array(
        "name" => "Mysql Connect",
        "description" => "Connect mysql server " . $aDb['server'] . " as user " . $aDb['username'] . " to scheme " . $aDb['database'],
        "check" => array(
            "function" => "MysqlConnect",
            "params" => array(
                "server"   => $aDb['server'],
                "user"     => $aDb['username'],
                "password" => $aDb['password'],
                "db"       => $aDb['database'],
                "port"     => $aDb['port'],
            ),
        ),
    )
);

if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS']){
    $oMonitor->addCheck(
        array(
            "name" => "Certificate check",
            "description" => "Check if SSL cert is valid and does not expire soon",
            "check" => array(
                "function" => "Cert",
            ),
        )
    );
}

// ----------------------------------------------------------------------

$oMonitor->setResult();
$oMonitor->render();

// ----------------------------------------------------------------------
